modules/filesystem/bm.nu

```diff
@@ -0,0 +1,83 @@
+# simple bookmark module
+
+# List all bookmarked paths
+export def list [] {
+  let bm_path = (get_path)
+
+  if (not ($bm_path | path exists))  {
+    [{name: "prev", path: ("~/" | path expand) }] | save $bm_path
+  }
+    open ($bm_path)
+}
+
+def get_path [] {
+  $env.XDG_DATA_HOME | 
```

Comment:
Can we have a default in case XDG_DATA_HOME doesn't exist?

Reply:
@Tiggax
i typically use the following construct to get a path / variable
```bash
def nupm-home [] {
    $env.NUPM_HOME? | default (
        $env.XDG_DATA_HOME?
        | default ($env.HOME | path join ".local" "share")
        | path join "nupm"
    )
}
```

Reply:
such that the application uses, in order,
- the user value
- the XDG value
- a default backup value

Reply:
Sure!, but what should it default to?
Im thinking:
-  `$env.HOME`, but that isn't good practice to clutter the home
- `$nu.temp-path` but that will wipe with every reboot
- `$nu.default-config-dir`

Reply:
mm i would rather
- `$env.BOOKMARKS_FILE`
- `$env.XDG_DATA_HOME` and then something like `nushell/bookmarks.nuon`
- `~/.local/share/` and then something like `nushell/bookmarks.nuon`

Reply:
to be more precise i would say no to
- `$env.HOME` because cluttering home is bad practice imo
- `$nu.temp-path` because it's not permanent
- `$nu.default-config-dir` because it's not really a config file => and the user can always change the path to make it so if they want to version the bookmarks for instance :yum:

Reply:
> mm i would rather
> 
>     * `$env.BOOKMARKS_FILE`
> 
>     * `$env.XDG_DATA_HOME` and then something like `nushell/bookmarks.nuon`
> 
>     * `~/.local/share/` and then something like `nushell/bookmarks.nuon`

I replied to fdncred, but my browser didn't update your comment :sweat_smile: . i changed it to your suggestion.

Reply:
oooooh i see :laughing: 

no harm done :wink:

Reply:
if Darren is ok with this new `get-path` then we can close this thread :yum: